### ISSUE TITLE
"Fatal Error: 'Uncaught Error: Cannot instantiate interface" after upgrade to 4.1.6

### ISSUE THREAD

**Issue by bamboekeizer** [myparcelnl/magento#574]
Every time cron is run, the error Fatal Error: 'Uncaught Error: Cannot instantiate interface is mailed and logged:

Email contains:
Cron <u10998d25711@web0104> /opt/alt/php73/usr/bin/php /private_html/bin/magento cron:run | grep -v "Ran jobs by schedule" >> /private_html/var/log/magento.cron.log

"Fatal Error: 'Uncaught Error: Cannot instantiate interface Magento\\InventoryApi\\Api\\SourceItemRepositoryInterface in \/public_html\/vendor\/magento\/framework\/ObjectManager\/Factory\/Dynamic\/Developer.php:50\nStack trace:\n#0 \/public_html\/vendor\/magento\/framework\/ObjectManager\/ObjectManager.php(70): Magento\\Framework\\ObjectManager\\Factory\\Dynamic\\Developer->create('Magento\\\\Invento...')\n#1 \/public_html\/vendor\/magento\/framework\/ObjectManager\/Factory\/AbstractFactory.php(167): Magento\\Framework\\ObjectManager\\ObjectManager->get('Magento\\\\Invento...')\n#2 \/public_html\/vendor\/magento\/framework\/ObjectManager\/Factory\/AbstractFactory.php(273): Magento\\Framework\\ObjectManager\\Factory\\AbstractFactory->resolveArgument(Array, 'Magento\\\\Invento...', NULL, 'sourceItemRepos...', 'MyParcelNL\\\\Mage...')\n#3' in '\/public_html\/vendor\/magento\/framework\/ObjectManager\/Factory\/Dynamic\/Developer.php' on line 50"

 - MyParcel Plugin version 4.1.6
 - MyParcel SDK version 5.2.10

**Comment by EdieLemoine** [comment]
Hi @bamboekeizer, could you share the full error? It should be in `private_html/var/log/magento.cron.log`.

**Comment by bamboekeizer** [comment]
That file is empty. I get the error emailed because of the cron setup.

**Comment by bamboekeizer** [comment]
I removed and added the myparcel and sdk modules again. The same issue remains. 
The side effect is that the user cannot complete the checkout anymore. When the user hits the payment button to complete the transaction, the checkout fails with an empty error message at the payment section. The error message lasts about a second and then disappears.




![Schermafbeelding 2021-06-15 090510](https://user-images.githubusercontent.com/28304222/122008095-eaafd300-cdb8-11eb-87cf-8b48c3b85631.png)

**Comment by bartvansaase** [comment]
@EdieLemoine I've the same issues as described here by @bamboekeizer. We have tried to install the module on a clean Magento env, in here we see the following errors from the logs when we try to place an order:

`[2021-06-15 12:57:28] main.CRITICAL: Saving order 000000001 failed: Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\Magento\Model\Source\SourceItem class [] []
[2021-06-15 12:57:28] main.CRITICAL: Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\Magento\Model\Source\SourceItem class {"exception":"[object] (ReflectionException(code: 0): Impossible to process constructor argument Parameter #1 [ <required> Magento\\InventoryApi\\Api\\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\\Magento\\Model\\Source\\SourceItem class at /home/358234.cloudwaysapps.com/xbhmczjknj/public_html/vendor/magento/framework/Code/Reader/ClassReader.php:45, ReflectionException(code: 0): Class Magento\\InventoryApi\\Api\\SourceItemRepositoryInterface does not exist at /home/358234.cloudwaysapps.com/xbhmczjknj/public_html/vendor/magento/framework/Code/Reader/ClassReader.php:34)"} []
[2021-06-15 12:59:36] main.CRITICAL: Saving order 000000002 failed: Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\Magento\Model\Source\SourceItem class [] []
[2021-06-15 12:59:36] main.CRITICAL: Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\Magento\Model\Source\SourceItem class {"exception":"[object] (ReflectionException(code: 0): Impossible to process constructor argument Parameter #1 [ <required> Magento\\InventoryApi\\Api\\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\\Magento\\Model\\Source\\SourceItem class at /home/358234.cloudwaysapps.com/xbhmczjknj/public_html/vendor/magento/framework/Code/Reader/ClassReader.php:45, ReflectionException(code: 0): Class Magento\\InventoryApi\\Api\\SourceItemRepositoryInterface does not exist at /home/358234.cloudwaysapps.com/xbhmczjknj/public_html/vendor/magento/framework/Code/Reader/ClassReader.php:34)"} []`

**Comment by RichardPerdaan** [comment]
Hi @bamboekeizer and @bartvansaase,
Do you use Magento's multi-source inventory?

Today I spent all day trying to reproduce the problem but without success.

**Comment by bartvansaase** [comment]
@RichardPerdaan No this has been disabled, if you want I can share access to our test environment. I've already send a mail also, you you can reply to this I can also share SSH access to our dev server so you can test it also (and debug).

**Comment by gjportegies** [comment]
We're also seeing in the magento.cron.log:

```
PHP Fatal error:  Uncaught Error: Cannot instantiate interface Magento\InventoryApi\Api\SourceItemRepositoryInterface in /home/domains/<domainname>/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:121
Stack trace:
#0 /home/domains/<domainname>/vendor/magento/framework/ObjectManager/Factory/Compiled.php(108): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('Magento\\Invento...', Array)
#1 /home/domains/<domainname>/vendor/magento/framework/ObjectManager/Factory/Compiled.php(150): Magento\Framework\ObjectManager\Factory\Compiled->create('Magento\\Invento...')
#2 /home/domains/<domainname>/vendor/magento/framework/ObjectManager/Factory/Compiled.php(79): Magento\Framework\ObjectManager\Factory\Compiled->get('Magento\\Invento...')
#3 /home/domains/<domainname>/vendor/magento/framework/ObjectManager/ObjectManager.php(70): Magento\Framework\ObjectManager\Factory\Compiled->create('MyParcelNL\\Mage...')
#4 /home/domains/<domainname>/vendor/myparcelnl/magento/Model/Sa in /home/domains/<domainname>/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 121
```

**Comment by RichardPerdaan** [comment]
We are currently investigating this issue, as soon as we know more I will post it here.

**Comment by Mitchell-MyParcel** [comment]
We cannot reproduce this issue on our test environments. If anyone can send us credentials (including SSH) for a test environment with this issue, that would be really helpful.

**Comment by bartvansaase** [comment]
@Mitchell-MyParcel I've send credentials to our test environment last friday to @RichardPerdaan.

**Comment by RichardPerdaan** [comment]
The problem has been found and I am now working on solving it. Once I have the solution I will share it with you.

**Comment by RichardPerdaan** [comment]
I have opened the following PR (#577), which will solve this problem.

You can try the adjustment to see if the issues have been resolved.
If it still doesn't work, then I would like to hear from you

**Comment by bamboekeizer** [comment]
I tried with the updated file and now the payment completes and there are no cron error messages. Thanks for the fix.

Btw, I did not remove multi inventory code

**Comment by RichardPerdaan** [comment]
Fixed in version[ 4.1.7](https://github.com/myparcelnl/magento/releases/tag/4.1.7) of the MyParcel Magento 2 plugin

**Comment by laurnts** [comment]
I also have the same issue with both 4.1.6 and 4.1.7

```
Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\  
  Magento\Model\Source\SourceItem class
```
